src/PhpOption/Option.php

```diff
@@ -99,6 +99,37 @@ public static function fromReturn($callback, array $arguments = array(), $noneVa
     }
 
     /**
+     * Option factory, which creates new option based on passed value.
+     * If value is already an option, it simply returns
+     * If value is a \Closure, LazyOption with passed callback created and returned. If Option returned from callback,
+     * it returns directly (flatMap-like behaviour)
+     * On other case value passed to Option::fromValue() method
+     *
+     * @param Option|\Closure|mixed $value
+     * @param null $noneValue used when $value is mixed or Closure, for None-check
+     *
+     * @return Option
+     */
+    public static function make($value, $noneValue = null)
+    {
+        if ($value instanceof Option) {
+            return $value;
+        } elseif ($value instanceof \Closure) {
+            return new LazyOption(function() use ($value, $noneValue) {
+                $return = $value();
+
+                if ($return instanceof Option) {
```

Comment:
I thought about recursivelly call make here. How about situation, when Closure is a expected value for Option result? If so, user should write somehing like:
```php
Option::make(function() {
   Some::create(function() {});
});
```

On other hand, we can explicitly call make inside Closure (on current implementation):
```php
Option::make(function() {
  return Option::make(function() {
    // extra lazy calculations
  });
});
```

What is preferred behaviour?